src/controllers/wallet/atomic-swap/tx-proposal.controller.js

```diff
@@ -126,13 +126,42 @@ async function buildTxProposal(req, res) {
     }
 
     let createdProposalId;
-    if (constants.SWAP_SERVICE_FEATURE_TOGGLE && serviceParams.is_new) {
-      const { proposalId } = await atomicSwapService.serviceCreate(
-        req.walletId,
-        proposal.partialTx.serialize(),
-        serviceParams.password
-      );
-      createdProposalId = proposalId;
+    if (constants.SWAP_SERVICE_FEATURE_TOGGLE) {
+      if (serviceParams.is_new) {
+        // Handling the creation of a new proposal with the Atomic Swap Service
+        const { proposalId } = await atomicSwapService.serviceCreate(
+          req.walletId,
+          proposal.partialTx.serialize(),
+          serviceParams.password
+        );
+        createdProposalId = proposalId;
+      } else if (serviceParams.proposal_id) {
+        // Handling the update of an existing proposal with the Atomic Swap Service
+
+        // First, validate if the proposal is already registered with this wallet
+        const listenedProposals = await atomicSwapService.getListenedProposals(req.walletId);
+        if (!listenedProposals.has(serviceParams.proposal_id)) {
+          res.status(404);
+          res.send({
+            success: false,
+            error: 'Proposal is not registered. Register it first through [POST] /register/:proposalId',
```

Comment:
The URL is wrong.
Also, I don't think we should use error messages to point to another API.

Reply:
Removed the URL on [c498c00](https://github.com/HathorNetwork/hathor-wallet-headless/pull/286/commits/c498c00478a141f46e8593dc54df0da66294fc92)